public/Leo Jayachandran/Glossarizer/libretextsglossarizer.js

```diff
@@ -0,0 +1,629 @@
+                        ['<td data-th="Word' + '(s)">', "word"],
+                        ['<td data-th="Definition">', "definition"],
+                        ['<td data-th="Exclusions">', "exclusion"],
+                        ['<td data-th="Image">', "image"],
+                        ['<td data-th="Caption">', "caption"],
+                        ['<td data-th="Link">', "link"],
+                        ['<td data-th="Source">', "source"]
+                    ]
+                    let colEnd = '</td>';
+                    for (let t = 0; t < colStart.length; t++) {
+                        let tag = colStart[t][0];
+                        let colStr = tableRows[r].substring(tableRows[r].search(tag) + tag.length);
+                        cols[colStart[t][1]] = (colStr.substring(0, colStr.search(colEnd)).trim());
+                    }
+                    //Generate the description
+                    let exclusions;
+                    if (cols["exclusion"].length) {
+                        exclusions = ", " + (cols["exclusion"].toLowerCase().replace(/<p>/g, " ").replace(/<\/p>/g, " ").trim().split(",")
+                            .map((val) => " !" + val.trim()).toString().trim());
+                    } else {
+                        exclusions = "";
+                    }
+                    newTerm["term"] = (cols["word"].substring(1).toLowerCase().replace(/<p>/g, " ").replace(/<\/p>/g, " ").trim() + exclusions).trim();
+
+                    //Make Description
+                    if (cols["link"].length) {
+                        let aTagStart = 'href="';
+                        let aTagEnd = '">';
+                        let href = cols["link"].substring(cols["link"].search(aTagStart) + aTagStart.length, cols["link"].search(aTagEnd));
+                        cols["definition"] = `<a href = "${href}" target="_blank">${cols["definition"]}</a>`;
+                    }
+                    if (cols["image"].length) {
+                        cols["definition"] += `<div class='imageContainer'>${cols["image"]}</div>`;
+                    }
+                    if (cols["caption"].length) {
+                        cols["definition"] += `<p class = 'caption'>${cols["caption"]}</p>`;
+                    }
+                    if (cols["source"].length) {
+                        cols["definition"] = cols["definition"].trim() + `<p class = "glossarySource">[Source: ${cols["source"].replace(/<p>/g, " ").replace(/<\/p>/g, " ").trim()}]</p>`;
+                    }
+
+                    newTerm["description"] = cols["definition"];
+
+                    //Add new term
+                    retrievedGlossary.push(newTerm);
+                }
+                return retrievedGlossary;
+            }
+        };
+        let chosenSource = sourceOption || "textbook";
+        retrievedGlossary = await glossaryRetriever[chosenSource]();
+        if (retrievedGlossary.length <= 1) { // Deal with incompatible Glossary
```

Comment:
Will this break if the glossary only has one term?

Reply:
It will break it, but at that expense, the glossary is able to catch a larger variety of problem cases, (usually if the glossarizer fails to get a term, it spits out a 1 term glossary with gibberish) so I thought it's a worthwhile tradeoff. If this does become a problem because we notice a lot of people using 1 term glossaries (which I suspect should still be relatively unlikely) then I can make the adjustment here.